src/compiler/scala/tools/nsc/typechecker/MethodSynthesis.scala

```diff
@@ -125,14 +125,21 @@ trait MethodSynthesis {
     def enterGetterSetter(tree: ValDef): Unit = {
       val fieldSym =
         if (noFieldFor(tree, owner)) NoSymbol
-        else owner.newValue(tree.name append NameTransformer.LOCAL_SUFFIX_STRING, tree.pos, tree.mods.flags & FieldFlags | PrivateLocal)
+        else owner.newValue(tree.name append NameTransformer.LOCAL_SUFFIX_STRING, tree.pos, tree.mods.flags & (FieldFlags | DEFERRED) | PrivateLocal)
```

Comment:
Oh, I forgot to explain that change in the commit comment. 

This isn't adding the `DEFERRED` flag, but rather adding it to the set of flags that are copied from the tree to the symbol. Abstract vals in traits were not being emitted in code gen without this change. I'll collect a trace of what was going wrong to justify this.